test/task_test.py

```diff
@@ -106,6 +109,28 @@ def test_no_unpicklable_properties(self):
         message = task.set_status_message('message')
         self.assertEqual(message, 'message')
 
+    def test_no_warn_if_param_types_ok(self):
+        with warnings.catch_warnings(record=True) as w:
+            DummyTask(**DUMMY_TASK_OK_PARAMS)
+        self.assertEqual(len(w), 0, msg='No warning should be raised when correct parameter types are used')
+
+    if six.PY3:  # assertWarnsRegex was introduced in Python 3.2
+        def test_warn_on_non_str_param(self):
+            params = dict(**DUMMY_TASK_OK_PARAMS)
```

Comment:
This test checks that an integer value (42) given to the "param" parameter (that expects a string) triggers a warning.
Hence I need to modify a copy of `DUMMY_TASK_OK_PARAMS`.